Catalyst-refactor-docQA (#946)
osisoft/OCS-Docs: content/analytics/data-views/data-views-retrieve.md

---
uid: data-views-retrieve
---

# Retrieve a data view

[!include[data-views-retrieve](../../_includes/data-views-retrieve.md)]

## Data retrieval options

To retrieve a data view, you can either use the OSIsoft Microsoft Power BI Plugin, or you can develop your own plugin using an example GitHub project as a starting point.

### Microsoft Power BI

> You can retrieve a data view using Microsoft Power BI. Retrieving a data view in Microsoft Power BI requires installation of OCS Power BI Connector. You can download the Connector by clicking on the banner listed at the top of the **Data Views** page. If you dismiss the banner, it will return upon your next user session.

Author: ```suggestion
You can retrieve a data view using Microsoft Power BI. Retrieving a data view in Microsoft Power BI requires installation of OCS Power BI Connector. The banner at the top of the [Data Views](xref:CreateDataView) page contains a link to the Connector.  If you dismiss the banner, it will return upon your next user session.
```
Can we also just link the Connector here?

Reviewer: I took the suggestion, but declined adding a link to the connector. Linking to things in Salesforce is a nightmare. I'll explain more during the next hootenanny.

> For more information installing the connector and retrieving a data view, see <xref:PowerBI>.

Author: ```suggestion
For more information about how to install the connector and retrieve a data view, see <xref:PowerBI>.
```

### Develop your own connector

Alternatively, you can develop your own connector for third-party software. AVEVA offers sample projects on GitHub for the following technologies:

- [Jupyter](https://github.com/osisoft/sample-adh-data_views_jupyter-python)
- [Python](https://github.com/osisoft/sample-adh-data_views-python)
- [R](https://github.com/osisoft/sample-adh-data_views_r-r)
- [.NET](https://github.com/osisoft/sample-adh-data_views-dotnet)
- [Java](https://github.com/osisoft/sample-adh-data_views-java)

## Workflow: Data view retrieval

To retrieve a data view in a third-party application, follow these steps:

### Step 1. Create Client-Credentials Client

> To retrieve a data view from OSIsoft Cloud Services (OCS), you must authenticate using a valid Client Id and Client secret. You can obtain an Id and secret by creating a set of client-credentials. While creating the client-credentials, accept the default expiry of 3600 seconds. When you are presented with the client secret, keep it secure, as there is no way to see the secret again. If you lose it, create a new set of client-credentials.

Author: ```suggestion
To retrieve a data view from OSIsoft Cloud Services (OCS), you must authenticate with a valid Client ID and Client secret. You can obtain an ID and secret by creating a set of client-credentials. When you create the client-credentials, accept the default expiry of 3600 seconds. When you receive the client secret, keep it secure, as there is no way to see the secret again. If you lose it, you need to create a new set of client-credentials.
```

For more information on creating client-credential clients, see <xref:gpClientCredentialsClient>.

> ### Step 2. Develop your own connector 

Author: For this step, could it also be "use the MS Power BI?" Maybe we can move that information into this step instead of having it above? Thoughts?

Reviewer: Agreed. I'll update.

Reviewer: Yeah, some info about MS was missing there. Thanks for pointing that out!
